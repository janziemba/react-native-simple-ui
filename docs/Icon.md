# Icon

## Props
- `color` (_string_, default null)
- `colorName` (_string_, default null)
- `iconSet` (_string_, default `theme.typography.iconSet` which is MaterialIcons by default)
- `name` (_string_, **required**)
- `size` (_number_, default 24)

## Examples

```jsx

```
