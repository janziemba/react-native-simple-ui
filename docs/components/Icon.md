---
id: icon
title: Icon
---
## Props
- `color` (_string_, default null)
- `colorName` (_string_, default null) - one of theme colors, [base theme colors](../Theme.md#colors) by default
- `iconSet` (_string_, default `theme.typography.iconSet` which is MaterialIcons by default)
- `name` (_string_, **required**)
- `size` (_number_, default 24)

## Examples

```jsx

```
