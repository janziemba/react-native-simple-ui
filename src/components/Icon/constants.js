import AntDesign from 'react-native-vector-icons/AntDesign';
import Entypo from 'react-native-vector-icons/Entypo';
import EvilIcons from 'react-native-vector-icons/EvilIcons';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Foundation from 'react-native-vector-icons/Foundation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';
import Zocial from 'react-native-vector-icons/Zocial';

import colors from '../../themes/base/colors';

export const COLORS = {
    ...colors.palette,
    ...colors.social,
    black: 'black',
    white: 'white',
};

export const ICON_SETS = {
    AntDesign,
    Entypo,
    EvilIcons,
    Feather,
    FontAwesome,
    Foundation,
    Ionicons,
    MaterialCommunityIcons,
    MaterialIcons,
    Octicons,
    SimpleLineIcons,
    Zocial,
};
