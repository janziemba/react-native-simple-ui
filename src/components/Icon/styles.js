// @flow

import merge from 'lodash/merge';

import type { ObjectOfStringsType, ObjectOfStyleSheetsType, ThemeShapeType } from '../../types';
import { COLORS } from './constants';

export type StylesType = {
    colors: ObjectOfStyleSheetsType,
};

export default ({ colors }: ThemeShapeType): StylesType => {
    const allColors: ObjectOfStringsType = merge({}, COLORS, colors.palette, colors.social);
    const colorOptions: ObjectOfStyleSheetsType = {};

    Object.keys(allColors).forEach((colorName: string) => {
        const color = allColors[colorName];

        colorOptions[colorName] = {
            color,
        };
    });

    return {
        colors: colorOptions,
    };
};
