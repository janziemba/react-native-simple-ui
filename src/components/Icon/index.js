// @flow

import React, { PureComponent } from 'react';
import type { Node } from 'react';

import withTheme from '../../themes/withTheme';
import type { ThemeShapeType } from '../../types';
import { ICON_SETS } from './constants';
import styles from './styles';
import type { StylesType } from './styles';

export type Props = {
    color?: string,
    colorName?: string,
    iconSet?: string,
    name: string,
    size?: number,
};

type PrivateProps = Props & {
    styles: StylesType,
    theme: ThemeShapeType,
};

class Icon extends PureComponent<PrivateProps> {
    static defaultProps = {
        color: null,
        colorName: null,
        iconSet: null,
        size: 24,
    };
    render(): Node {
        const { color, colorName, iconSet, name, size, styles, theme } = this.props;

        const VectorIcon = ICON_SETS[iconSet || theme.typography.iconSet];

        let iconColor: ?string = color;

        if (colorName) {
            iconColor = styles.colors[colorName].color;
        }

        return (
            <VectorIcon
                color={iconColor}
                name={name}
                size={size}
            />
        );
    }
}

Icon = withTheme(styles, 'Icon')(Icon);

export default Icon;
