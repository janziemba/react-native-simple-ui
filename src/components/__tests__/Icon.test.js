import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../Icon';

it('renders correctly', () => {
    const tree = renderer.create(
        <Icon name="favorite" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
