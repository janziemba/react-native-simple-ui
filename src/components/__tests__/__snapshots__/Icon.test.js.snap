// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": null,
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  î¡½
</Text>
`;
